cpp/oneapi/dal/algo/triangle_counting/backend/cpu/vertex_ranking_default_kernel.hpp

```diff
@@ -23,34 +23,250 @@
 #include "oneapi/dal/backend/interop/table_conversion.hpp"
 #include "oneapi/dal/detail/threading.hpp"
 #include "oneapi/dal/table/detail/table_builder.hpp"
+#include "oneapi/dal/algo/triangle_counting/backend/cpu/intersection_tc.hpp"
+#include "oneapi/dal/backend/primitives/intersection/intersection.hpp"
 
 namespace oneapi::dal::preview::triangle_counting::backend {
 
 template <typename Cpu>
-std::int64_t triangle_counting_global_scalar(const std::int32_t* vertex_neighbors,
-                                             const std::int64_t* edge_offsets,
-                                             const std::int32_t* degrees,
-                                             std::int64_t vertex_count,
-                                             std::int64_t edge_count);
+array<std::int64_t> triangle_counting_local(const dal::preview::detail::topology<std::int32_t>& t,
+                                            int64_t* triangles_local) {
```

Comment:
what is the connection between returned array and triangulars_local?

Reply:
it is an array that will be underneath result's table

Reply:
It was moved to this separate function as both local and local_and_global tasks need to earn this array.